core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1812,6 +1812,13 @@ class SparkContext(config: SparkConf) extends Logging {
   /** The version of Spark on which this application is running. */
   def version: String = SPARK_VERSION
 
+  /**
+   * Return an array of executors' host name from the block manager.
+   */
+  def getExecutorHosts: Array[String] = {
```

Comment:
I wouldn't add this as an API at `SparkContext`. You could just do it with `getExecutorMemoryStatus.map(_._1).toArray`.

Reply:
Thanks for your comment. Your comment just remind me there is a typo-error in previous commit. It should be:
getExecutorMemoryStatus.map(_._1.split(":")(0)).toArray

It is not an api in my first thought. But the splitting here is kind of messy in other place. Making it as an api would help me out, and would enclose implementation of how we get the host name list.